website/server/controllers/api-v3/tasks.js

```diff
@@ -413,9 +413,13 @@ api.updateTask = {
       throw new NotFound(res.t('taskNotFound'));
     } else if (task.group.id && !task.userId) {
       //  @TODO: Abstract this access snippet
-      group = await Group.getGroup({user, groupId: task.group.id, fields: requiredGroupFields});
+      let fields = requiredGroupFields.concat(' managers');
+      group = await Group.getGroup({user, groupId: task.group.id, fields});
       if (!group) throw new NotFound(res.t('groupNotFound'));
-      if (group.leader !== user._id) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
+      let isNotGroupLeader = group.leader !== user._id;
```

Comment:
Can `isNotGroupLeader` be changed to `isGroupLeader`? That would be consistent with `isManager`.